### PR TITLE
Standardize colors with SCSS variables

### PR DESCRIPTION
**Pull request by hanywang2**
### Summary
The colors used for CoursePlan was super decentralized. Every instance of using colors with hex codes is just copy-pasted directly from the designs. This PR is a massive overhaul to centralize the colors with SCSS variables.

The variables can be seen at and changed at `src/assets/scss/_variables.scss`.

**Instructions to import global color variables** 
```JS
// To import into <script>
import { inactiveBlue, ... } from '@/assets/scss/_variables.scss';

// To import into <style>
@import "@/assets/scss/_variables.scss";
```

### Test Plan
All color schemes should work as before and color names consistent with the designs defined here.
Screen Shot 2020-10-24 at 7 06 00 PM

Screen Shot 2020-10-24 at 7 03 29 PM
Screen Shot 2020-10-24 at 7 03 19 PM
Screen Shot 2020-10-24 at 7 03 10 PM
Screen Shot 2020-10-24 at 7 03 00 PM

This pull request is predominately a refactoring effort involving CSS and minor changes in JS (all involving the altering of Hex codes). Nothing should be broken as a result.

### CODE REVIEW

File: src/assets/scss/_variables.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+$medium: #657786;
+$dark: #34495e;
+$white: #fff;
+$black: #000;
+
+$offWhite: #f8f8f8;
+$medGray: #858585;
+$darkGray: #3C3C3C;
+
+$emGreen: #1AA9A5;
```
Comment (einc) [inline src/assets/scss/_variables.scss:14]: sorry Yuxuan & I made this color darker to pass contrast ratio tests but forgot to update to Figma, it should be #148481

File: src/assets/scss/_variables.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+$chrisGreen: #105351;
+$sangBlue: #508197;
```
Comment (einc) [inline src/assets/scss/_variables.scss:16]: sangBlue should be #4D7D92

File: src/assets/scss/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+$einBlue: #92C3E6;
+
+$inactiveBlue: #32A0F2;
```
Comment (einc) [inline src/assets/scss/_variables.scss:19]: can we call this yuxuanBlue 🥺🥰 so each designer gets a blue color haha, thank you!

**Review by einc — COMMENTED** [review]
Changing two of the brand colors to meet accessibility guidelines

File: src/assets/scss/_variables.scss (hunk 3 above)
Comment (hanywang2) [reply src/assets/scss/_variables.scss]: Updated!

**Comment by tcho6319** [comment]
Wow thanks for this overhaul Han! In `src/assets/images` we have `.svg` files like `dropup-lightblue.svg` which now might be one of the SCSS variable names. Have you considered renaming those .svg files to correspond with your changes? Though I should note that in `src/assets/components/Requirements.vue`, we have `reqGroupColorMap` which would have to change if we changed the names of the `.svg` files:
```
// reqGroupColorMap maps reqGroup to an array [<hex color for progress bar>, <color for arrow image>]
      reqGroupColorMap: {
        UNIVERSITY: ['508197', 'grayblue'],
        COLLEGE: ['1AA9A5', 'blue'],
        MAJOR: ['105351', 'green'],
        MINOR: ['92C3E6', 'lightblue']
      }
```

**Review by tcho6319 — COMMENTED** [review]
Wow thanks for this overhaul Han! In `src/assets/images` we have `.svg` files like `dropup-lightblue.svg` where lightblue now might be one of the SCSS variable names. Have you considered renaming those `.svg` files to correspond with your changes? Though I should note that in `src/assets/components/Requirements.vue`, we have `reqGroupColorMap` which would have to change if we changed the names of the `.svg` files:
```
// reqGroupColorMap maps reqGroup to an array [<hex color for progress bar>, <color for arrow image>]
      reqGroupColorMap: {
        UNIVERSITY: ['508197', 'grayblue'],
        COLLEGE: ['1AA9A5', 'blue'],
        MAJOR: ['105351', 'green'],
        MINOR: ['92C3E6', 'lightblue']
      }
```
Maybe these filenames could be refactored in this PR or another one?

**Review by willespencer — CHANGES_REQUESTED** [review]
Thanks for doing this Han, it looks great 😄. 

I noticed that the edit color menu does not show up anymore, probably as a result of these changes? I think if you fix up that bug and maybe make @tcho6319's suggested improvements, this should be good to go.

**Comment by hanywang2** [comment]
@willespencer fixed the issue. It was really small thing. Sorry I couldn't have done it earlier

**Comment by hanywang2** [comment]
Screen Shot 2020-10-30 at 2 02 29 PM

**Review by willespencer — APPROVED** [review]
Haha that was a really small fix, looks good! Please just make a low priority Monday item for what Theresa mentioned (in Codebase fixes?) and then you're good to merge 😄

**Pull request merged**